### PR TITLE
Unread count to badge mode

### PR DESCRIPTION
**Pull request by lemieux**
The title says it all.

image

I'll add some basic tests for it.

@chloepouprom @dannytranlx @Mario54 @jugarrit @alavers

### CODE REVIEW

**Comment by dannytranlx** [comment]
beauty pr 👌 
👍

File: src/js/components/badge.jsx [hunk 1]
```diff
@@ -39,17 +40,19 @@ export class DefaultBadgeIcon extends Component {
...
 
-export class Badge extends Component {
+export class BadgeComponent extends Component {
     static contextTypes = {
         settings: PropTypes.object.isRequired
     };
 
     static propTypes = {
-        shown: PropTypes.bool
+        shown: PropTypes.bool,
```
Comment (chloepouprom) [inline src/js/components/badge.jsx]: Should this be a required proptype?

Comment (lemieux) [inline src/js/components/badge.jsx]: there's a default prop value for it so no.

**Comment by mspensieri** [comment]
👍 

too much badge

File: test/specs/components/badge.spec.js [hunk 2]
```diff
@@ -44,9 +44,36 @@ describe('Badge Component', () => {
...
+
+        const context = {
+            settings: {}
+        };
+
+        const component = wrapComponentWithContext(BadgeComponent, props, context);
+        TestUtils.scryRenderedDOMComponentsWithClass(component, 'unread-badge').length.should.eq(0);
+    });
+
+    it('should render unread count if none', () => {
```
Comment (chloepouprom) [inline test/specs/components/badge.spec.js]: this isn't what the test is doing

Comment (lemieux) [inline test/specs/components/badge.spec.js]: ehhh bad copy paste !

**Comment by chloepouprom** [comment]
looks good! 👍

File: src/js/components/badge.jsx [hunk 3]
```diff
@@ -58,30 +61,43 @@ export class Badge extends Component {
...
+            content = <div className='badge-icon'>
+                          <img src={ badgeIconUrl } />
+                      </div>;
         } else {
             content = <DefaultBadgeIcon isBrandColorDark={ isBrandColorDark }
-                                   brandColor={ brandColor } />;
+                                        brandColor={ brandColor } />;
+        }
+
+        let unreadBadge;
```
Comment (alavers) [inline src/js/components/badge.jsx]: The unread count is clearly a badge, but the chat button isn't really a badge at all in this sense. Unfortunate (hindsight is 20/20) naming ¯\_(ツ)_/¯

**Comment by alavers** [comment]
👍

File: src/js/components/badge.jsx (hunk 3 above)
Comment (lemieux) [inline src/js/components/badge.jsx]: Meh, that's an unread badge on a chat badge for your page.

**Pull request merged**